airflow/operators/python.py

```diff
@@ -370,12 +375,8 @@ def _read_result(self, path: Path):
             return None
         try:
             return self.pickling_library.loads(path.read_bytes())
-        except ValueError:
-            self.log.error(
-                "Error deserializing result. Note that result deserialization "
-                "is not supported across major Python versions."
-            )
-            raise
+        except ValueError as e:
+            raise DeserializingResultError(f"Error while deserializing results. Error: {e}")
```

Comment:
Isn’t this message useless since you overriden `__str__` in the class?

Reply:
Good catch!